applications/tari_base_node/src/builder.rs

```diff
@@ -495,43 +493,155 @@ where
             };
 
             if bootstrapped {
+                debug!(
+                    target: LOG_TARGET,
+                    "Initial sync achieved and starting with transaction and UTXO validation protocols.",
+                );
                 let _ = transaction_service_handle
                     .restart_broadcast_protocols()
                     .await
                     .map_err(|e| {
                         error!(
                             target: LOG_TARGET,
-                            "Problem restarting broadcast protocols in the Transaction Service"
+                            "Problem restarting broadcast protocols in the Transaction Service: {}", e
                         );
                         e
                     });
-
                 let _ = transaction_service_handle
                     .restart_transaction_protocols()
                     .await
                     .map_err(|e| {
                         error!(
                             target: LOG_TARGET,
-                            "Problem restarting transaction negotiation protocols in the Transaction Service"
+                            "Problem restarting transaction negotiation protocols in the Transaction Service: {}", e
                         );
                         e
                     });
+
+                #[derive(PartialEq, Debug)]
+                enum ValidationTask {
```

Comment:
I think having an enum here in the method is a sign that this logic should be separated into simpler methods.

Reply:
Moving this logic out to a separate function as @philipr-za suggested.

Note that the enum is to prevent duplicating code from line 665 to 733, starting here.
``` rust
                loop {
                    futures::select! {
```

Reply:
I think the point is that with the correct use of a function the code would not be duplicated, but also I think you only need to monitor for the Unspent validation anyway.

Reply:
> I think the point is that with the correct use of a function the code would not be duplicated

Yes, I see the point now, this can be done